modules/javafx.controls/src/main/java/javafx/scene/control/MultipleSelectionModelBase.java

```diff
@@ -768,8 +768,9 @@ public void set(int index, int... indices) {
                         .filter(this::isNotSelected)
                         .sorted()
                         .boxed()
-                        .peek(this::set) // we also set here, but it's atomic!
                         .collect(Collectors.toList());
+
+                sortedNewIndices.forEach(this::set);
```

Comment:
Why do the double-iteration pattern here and not do the `peek` operation in a `forEach` like in the other 2 places?

Reply:
`forEach` is void, so we can not return a list afterwards.

Reply:
You don't need to return a list, you create it ahead of time like was done in line 167
```java
List<Integer> indices = new ArrayList<>();
```
and the add the elements in `forEach`.

Reply:
> Why do the double-iteration pattern here and not do the `peek` operation in a `forEach` like in the other 2 places?

Yes, if the usage of `forEach` in previous 2 places are correct then I would like to see similar change here as well.

Reply:
I changed this to the pre-allocated pattern like in the other places. An initial capacity of `indices.length + 1` optimizes the worst case (all given indices need to be set) which probably occurs quite often in real-world usage.